src/__internal__/form-field/form-field.component.tsx

```diff
@@ -43,7 +43,13 @@ interface CommonFormFieldProps extends MarginProps, ValidationProps {
   validationIconId?: string;
 }
 
-export interface FormFieldProps extends CommonFormFieldProps, TagProps {
+export interface FormFieldProps extends CommonFormFieldProps {
```

Comment:
It might be easier to just update TagProps to make the `data-component` type optional, like Maciek has done [here](https://github.com/Sage/carbon/blob/47fa084e2eb12ca80c2f17fd93bd15c59aa727cf/src/__internal__/utils/helpers/tags/tags.ts), then you can keep using the util